### PR TITLE
Add webservice to OrderCartRule with endpoint order_cart_rules

### LINKED ISSUE #13394: `order_discounts` not available via WebServices
**Describe the bug**
I have a connector Odoo <-> PrestaShop worked with PrestaShop 1.6.x. I tried to upgrade the connector to work with 1.7.x versions of PrestaShop, but with 1.7.x versions, the `order_discounts` is not available via WebServices, so my connector doesn't work.

**To Reproduce**
1. Go to `Advanced Parameters > WebServices`
2. Create a new key
3. In the Permissions settings, the `order_discounts` object is not available

**Additionnal information**
PrestaShop version: 1.7.x

### PR DESCRIPTION
**Pull request by Daxtermania**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Enable OrderCartRules in webservice
| Type?         | improvement 
| Category?     | WS
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #13394
| How to test?  | In multiple Prestashop version 1.7

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16710)

### CODE REVIEW

**Review by Daxtermania — COMMENTED** [review]
All is correct

File: classes/webservice/WebserviceRequest.php [hunk 1]
```diff
@@ -308,11 +308,12 @@ public static function getResources()
             'messages' => array('description' => 'The Messages', 'class' => 'Message'),
             'order_carriers' => array('description' => 'The Order carriers', 'class' => 'OrderCarrier'),
             'order_details' => array('description' => 'Details of an order', 'class' => 'OrderDetail'),
-            'order_histories' => array('description' => 'The Order histories', 'class' => 'OrderHistory'),
-            'order_invoices' => array('description' => 'The Order invoices', 'class' => 'OrderInvoice'),
-            'orders' => array('description' => 'The Customers orders', 'class' => 'Order'),
-            'order_payments' => array('description' => 'The Order payments', 'class' => 'OrderPayment'),
-            'order_states' => array('description' => 'The Order statuses', 'class' => 'OrderState'),
+            'order_histories' => array('description' => 'The Order histories','class' => 'OrderHistory'),
```
Comment (PierreRambaud) [inline classes/webservice/WebserviceRequest.php]: Please don't change all lines. Make sure to edit only the needed one. Here, you break the Php-cs-fixer test :)

**Review by PierreRambaud — APPROVED** [review]

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Could you move it to line 310?

**Comment by Daxtermania** [comment]
> Could you move it to line 310?

Ready! Thx!

**Review by PierreRambaud — DISMISSED** [review]

**Comment by Daxtermania** [comment]
What is the problem?

**Comment by Progi1984** [comment]
@Daxtermania Your PR need to pass QA tests before being merged. Could you rebase your PR for fixing the conflicting file ? Thanks

**Review by Progi1984 — DISMISSED** [review]

**Comment by Daxtermania** [comment]
All OK?

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by Daxtermania** [comment]
All OK?

**Comment by PierreRambaud** [comment]
@Daxtermania See @Progi1984's comment :wink: 

> @Daxtermania Your PR need to pass QA tests before being merged. Could you rebase your PR for fixing the conflicting file ? Thanks

**Comment by SD1982** [comment]
The order_cart_rules webservices is correctly added by this PR. Thanks @Daxtermania

**Comment by Progi1984** [comment]
Thanks @Daxtermania

**Comment by bodi000** [comment]
Hello,
Is there any reason for not pushing this to 1.7.6 for example? Thanks

**Comment by Uebix** [comment]
@PierreRambaud , @SD1982 please change the target 1.7.8.0, it is too far.
This is a regression fix, is impossible to wait for 1.7.8.
Is there any reason for not pushing this in the next release?
I know that 1.7.7.0 is in RC but this bugfix can be ready for 1.7.7.1 (for example).

**Comment by Progi1984** [comment]
@Uebix @bodi000 FYI, we are currently the stable version of 1770. So as the bug is not critical or major, it will be fixed in the next release.

**Comment by adrielalonso** [comment]
good evening. When do you plan to release version 1.7.8? I also need to access order_cart_rule through the api
Thanks

**Comment by chattago2002** [comment]
Hi. Is there the possibility to reintroduce the access to order_cart_rule through the api in the next release (1.7.7.6)?
PS 1.7.8.0 is too far for who needs to have this data for webservice usage.
Thanks

**Comment by matks** [comment]
> Hi. Is there the possibility to reintroduce the access to order_cart_rule through the api in the next release (1.7.7.6)?
> PS 1.7.8.0 is too far for who needs to have this data for webservice usage.
> Thanks

I'm sorry, no, this would be against our patch strategy.

You can apply the changes on your shop to benefit from the bugfix, but if there is a 1.7.7.6 release it will not contain this bugfix.

**Comment by chattago2002** [comment]
> > Hi. Is there the possibility to reintroduce the access to order_cart_rule through the api in the next release (1.7.7.6)?
> > PS 1.7.8.0 is too far for who needs to have this data for webservice usage.
> > Thanks
> 
> I'm sorry, no, this would be against our patch strategy.
> 
> You can apply the changes on your shop to benefit from the bugfix, but if there is a 1.7.7.6 release it will not contain this bugfix.

Ok, I will try on my own

**Pull request merged**